spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/integration/IntegrationProperties.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.integration;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.integration.http.support.HttpContextUtils;
+
+/**
+ * Configuration properties for Spring Integration.
+ *
+ * @author Artem Bilan
+ * @since 1.4
+ * @see org.springframework.integration.context.IntegrationProperties
+ */
+@ConfigurationProperties("spring.integration")
+public class IntegrationProperties {
+
+	private final Channels channels = new Channels();
+
+	/**
+	 * The number of threads available in the default taskScheduler.
+	 */
+	private int taskSchedulerPoolSize = 10;
+
+	/**
+	 * When {@code true}, messages that arrive at a gateway reply channel will throw an exception.
+	 */
+	private boolean throwExceptionOnLateReply = false;
+
+	/**
+	 * When {@code true}, Messaging Annotation Support requires a declaration of the {@code @MessageEndpoint}.
+	 */
+	private boolean componentAnnotationRequired = false;
```

Comment:
No-op. That was @garyrussell 's wish to avoid some breaking changes: https://github.com/spring-projects/spring-integration/pull/1760#issuecomment-208504119 :smile: 

We can reconsider that in SI-5.0 and yes, remove altogether and don't expose it as an auto-config option at all.